moncli/types.py

```diff
@@ -217,7 +180,79 @@ def _cast(self, value):
     def _export(self, value):
         return { 'rating': value}
 
-        
+
+class StatusType(MondayType):
+    native_type = str
+    allow_casts = (int,)
+    null_value = {}
+
+    def __init__(self, id: str = None, title: str = None, as_enum: type = None ,  *args, **kwargs):
+        if as_enum:
+            try:
+                enum_value = None
+                for value in list(as_enum):
+                    enum_value = value.name
+                    str(enum_value)
+                self._as_enum =  as_enum
+                self.native_type = Enum
+                self.allow_casts = str
+                self.choices = list(as_enum)
+            except TypeError:
+                raise ConversionError('Invalid value "{}" for status Enum "{}".'.format(enum_value, enum_value.__class__))    
+        super().__init__(id=id, title=title, *args, **kwargs)
+    
+    def _process(self, value):
+        labels = self.metadata['labels']
+        label = str(value)
+        if self.native_type != str:
+            return value
+        try:
+            value = int(value)
+            try:
+                return labels[label]
+            except KeyError:
+                raise ConversionError('Cannot find status label with index "{}".'.format(value))
+        except ValueError:
+            if value in labels.values():
+                return value
+            raise ConversionError('Cannot find status label "{}".'.format(value))
+                
+    def _cast(self, value):
+        labels = self.metadata['labels']
+        label = str(value)
+        if self.native_type == str and isinstance(value,int):
+            try:    
+                return labels[label]
+            except KeyError:
+                raise ConversionError('Cannot find status label with index "{}".'.format(value))
+            
+        if self.native_type == Enum and isinstance(value,str):
+            try:
+                return self._as_enum(label)
+            except ValueError:
+                    raise ConversionError('Cannot find status label with index "{}".'.format(value))
+    
+    def _process_column_value(self, column_value: en.cv.ColumnValue):
+        if self.native_type == str:
+            self.choices = [value for value in self.metadata['labels'].values()]
+                      
+    def _export(self, value):
```

Comment:
Lets simplify this method a little bit and remember the logical flow from top to bottom.  The flow is thus:

1. Find the label from whatever is passed in
2. Return the formatted export dict using the corresponding index value.

Keeping this in mind, we can rewrite the function as so.
```python
def _export(self, value):
        labels = self.metadata['labels']
        if self.native_type == str:
            label = value
        elif isinstance(self.native_type, EnumMeta):
            label = value.value
        for index, value in labels.items():
            if value == label:
                return {'index': int(index)}
```